drivers/can/can_mcp251xfd.c

```diff
@@ -30,7 +30,7 @@ LOG_MODULE_REGISTER(can_mcp251xfd, CONFIG_CAN_LOG_LEVEL);
 static void mcp251xfd_canframe_to_txobj(const struct can_frame *src, int mailbox_idx,
 					struct mcp251xfd_txobj *dst)
 {
-	dst->flags = 0;
+	memset(dst, 0, sizeof(*dst));
```

Comment:
The device will only the transmit the payload set by the DLC. I don't think we need zero to initialize the whole frame.

Reply:
Could we then just zero out upto the data field? (the way I did on the receiver side)

Reply:
No. The intent is to not leak uninitialized data from the ISR stack/kernel thread stack to the application/user mode thread - and vice versa.

Reply:
For the transmit side, we should modify the spi transfer length so that it doesn't include the data buffer when RTR flag is set. This way we don't have to zero initialize the data. If you want, I can send in a separate patch for this so that you don't have to get everyone to approve again.

Reply:
Btw I tested your patch on the mikroe mcp2518fd click and API test passes fine. (Although the receive rtr tests are skipped now because CAN_FILTER_RTR is not supported). I'll work on a patch to add this.

Reply:
> For the transmit side, we should modify the spi transfer length so that it doesn't include the data buffer when RTR flag is set. This way we don't have to zero initialize the data. If you want, I can send in a separate patch for this so that you don't have to get everyone to approve again.

@talih0 Yes, that sounds like a nice improvement. Please submit this as a separate PR and let's try to get that into v3.5.0 as well.